### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by yuran007**
[DEMO LINK](https://yuran007.github.io/react_todo-app-with-api/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,24 +1,312 @@
...
+  title,
+  userId: USER_ID,
+  completed: false,
+});
+const deleteTodoOnServer = (todoId: number) => client
+  .delete(`/todos/${todoId}`);
+const toggleTodoOnServer = (todoId: number, completed: boolean) => client
+  .patch<Todo>(`/todos/${todoId}`, { completed: !completed });
+const updateTodoOnServer = (todoId: number, title: string) => client
+  .patch<Todo>(`/todos/${todoId}`, { title });
```
Comment (sTorba24) [inline src/App.tsx:27]: it should be in `api/todos.ts` file
never create handlers for api in main App

File: src/App.tsx [hunk 2]
```diff
@@ -1,24 +1,312 @@
...
+    todosFromServer
+      .then(fetchedTodos => setTodos(fetchedTodos))
+      .catch(() => showErrors(ShowError.fetchTodos));
+  }, []);
+
+  const filteredTodos = React.useMemo(() => {
+    return getFilteredTodos(selectedTodos);
+  }, [selectedTodos, todos]);
+
+  const createTodo = (event: React.FormEvent) => {
```
Comment (sTorba24) [inline src/App.tsx:105]:
```suggestion
  const handleCreateTodo = (event: React.FormEvent) => {
```

Comment (sTorba24) [reply src/App.tsx:105]: follow handlers naming convention

File: src/App.tsx [hunk 3]
```diff
@@ -1,24 +1,312 @@
...
+    if (todoTitle === '') {
```
Comment (sTorba24) [inline src/App.tsx:106]:
```suggestion
    if (!todoTitle) {
```

File: src/App.tsx [hunk 4]
```diff
@@ -1,24 +1,312 @@
...
+      showErrors(ShowError.createTodo);
+    }
+
+    event.preventDefault();
+    if (todoTitle !== null) {
```
Comment (sTorba24) [inline src/App.tsx:111]:
```suggestion
    if (todoTitle) {
```

File: src/App.tsx (hunk 2 above)
Comment (sTorba24) [reply src/App.tsx:105]: fix everywhere

File: src/App.tsx [hunk 5]
```diff
@@ -1,24 +1,312 @@
...
+            )}
+
+          </footer>
+        )}
+      </div>
+
+      <div className={classNames(
+        'notification is-danger is-light has-text-weight-normal',
+        {
+          hidden: error === null,
```
Comment (sTorba24) [inline src/App.tsx:298]:
```suggestion
          hidden: !error,
```

File: src/api/todos.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import { Todo } from '../types/Todo';
+import { client } from '../utils/fetchClient';
+
+export const getTodos = (userId: number) => {
+  return client.get<Todo[]>(`/todos?userId=${userId}`);
+};
+
+// Add more methods here
```
Comment (sTorba24) [inline src/api/todos.ts:8]: it was hint that api functions should be here

File: src/components/TodoItem/TodoItem.tsx [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+  ({
+    todo,
+    deleteTodo,
+    updateTodo,
+  }) => {
+    const [editingTodo, setEditingTodo] = useState<Todo | null>(null);
+
+    const handleSubmit = (event: React.FormEvent) => {
+      event.preventDefault();
+      if (editingTodo === null) {
```
Comment (sTorba24) [inline src/components/TodoItem/TodoItem.tsx:21]:
```suggestion
      if (!editingTodo) {
```

**Review by sTorba24 — CHANGES_REQUESTED** [review]

**Review by sTorba24 — CHANGES_REQUESTED** [review]
Almost good! Lets improve it

File: src/App.tsx (hunk 1 above)
Comment (yuran007) [reply src/App.tsx:27]: Done

File: src/App.tsx (hunk 2 above)
Comment (yuran007) [reply src/App.tsx:105]: Done

File: src/App.tsx (hunk 3 above)
Comment (yuran007) [reply src/App.tsx:106]: Done, but I prefer to not use uses type conversion in boolean context.

File: src/App.tsx (hunk 4 above)
Comment (yuran007) [reply src/App.tsx:111]: Done

File: src/App.tsx (hunk 5 above)
Comment (yuran007) [reply src/App.tsx:298]: Done

File: src/api/todos.ts (hunk 6 above)
Comment (yuran007) [reply src/api/todos.ts]: I forgot to look at this file at all.

File: src/components/TodoItem/TodoItem.tsx (hunk 7 above)
Comment (yuran007) [reply src/components/TodoItem/TodoItem.tsx:21]: Done

**Comment by yuran007** [comment]
Please check animation also, thanks.

**Review by demetrxx — APPROVED** [review]
Well done 👍 Go ahead!

**Pull request closed without merging**